src/incubator/TextField/CharCounter.tsx

```diff
@@ -7,13 +7,14 @@ import {CharCounterProps} from './types';
 
 const CharCounter = ({maxLength, charCounterStyle, testID}: CharCounterProps) => {
   const {value} = useContext(FieldContext);
+  const length = useMemo(() => (_.isUndefined(value) ? 0 : value.length), [value]);
```

Comment:
Memoizing the length is a little overkill, also it's primitive value (number) so there no real reason to memoize it

Reply:
Also consider simplifying the code by doing something like this `value?.length ?? 0`